localization/twist_estimator/gyro_odometer/src/gyro_odometer_core.cpp

```diff
@@ -16,49 +16,52 @@
 
 #include "gyro_odometer/gyro_odometer_core.h"
 
-#include <cmath>
-
 #include <tf2_geometry_msgs/tf2_geometry_msgs.h>
 
-GyroOdometer::GyroOdometer(ros::NodeHandle nh, ros::NodeHandle private_nh)
-: nh_(nh), private_nh_(private_nh), output_frame_("base_link"), tf2_listener_(tf2_buffer_)
+#include <cmath>
+
+GyroOdometer::GyroOdometer()
+: Node("map_tf_generator"), 
+  output_frame_(declare_parameter("base_link", "base_link"))
 {
-  private_nh_.getParam("output_frame", output_frame_);
+  rclcpp::Clock::SharedPtr clock = std::make_shared<rclcpp::Clock>(RCL_ROS_TIME);
+  tf_buffer_ = std::make_shared<tf2_ros::Buffer>(clock);
```

Comment:
We could also make them plain objects instead of shared pointers. I find it nice to use plain objects by default, what do you think?

Reply:
Sure, I can do that. I was using smart pointers as this was how it was done in the rclcpp test code for these objects - though that's not really an excuse to use them. I can change this to use plain objects

Reply:
I'll leave it up to you.